### PR TITLE
reset graph slowlog

### PR DESCRIPTION
**Pull request by swilly22**
Reset graph's slowlog
`GRAPH.SLOWLOG G RESET`

### CODE REVIEW

File: src/commands/cmd_slowlog.c [hunk 1]
```diff
@@ -4,20 +4,60 @@
...
+	//--------------------------------------------------------------------------
+	// validations
+	//--------------------------------------------------------------------------
 
-	CommandCtx_TrackCtx(command_ctx);
+	ASSERT(ctx  != NULL);
+	ASSERT(argv != NULL);
+	if(argc < 2 || argc > 3) {
+		RedisModule_WrongArity(ctx);
+		return REDISMODULE_OK;
```
Comment (AviAvni) [inline src/commands/cmd_slowlog.c:29]: why return ok if it is error?

File: src/commands/cmd_slowlog.c [hunk 2]
```diff
@@ -4,20 +4,60 @@
...
+	}
+
+	// get a hold of the graph key
+	RedisModuleString *key = argv[1];
+	GraphContext *gc = GraphContext_Retrieve(ctx, key, false, false);
+	if(gc == NULL) {
+		// if GraphContext is null, key access failed and an error been emitted
+		return REDISMODULE_OK;
```
Comment (AviAvni) [inline src/commands/cmd_slowlog.c:37]: why return ok if it is error?

File: src/commands/cmd_slowlog.c [hunk 3]
```diff
@@ -4,20 +4,60 @@
...
+
+	// handle subcommand e.g. GRAPH.SLOWLOG G RESET
+	if(argc == 3) {
+		const char *sub_cmd = RedisModule_StringPtrLen(argv[2], NULL);
+		if(strcasecmp(sub_cmd, "reset") == 0) {
+			SlowLog_Clear(slowlog);
+			RedisModule_ReplyWithSimpleString(ctx, "OK");
+		} else {
+			// unknown subcommand
+			RedisModule_ReplyWithError(ctx, "Unknown subcommand");
```
Comment (AviAvni) [inline src/commands/cmd_slowlog.c:50]: in error you reply with REDISMODULE_OK

File: docs/commands/graph.slowlog.md [hunk 4]
```diff
@@ -18,3 +18,11 @@ GRAPH.SLOWLOG graph_id
     3) "MATCH (me:Person)-[:FRIEND]->(:Person)-[:FRIEND]->(fof:Person) RETURN fof.name"
     4) "0.288"
 ```
+
+To reset a graph's slowlog issue the following command:
+
+```sh
+GRAPH.SLOWLOG graph_id RESET
```
Comment (AviAvni) [inline docs/commands/graph.slowlog.md:25]: didn't we prefer 
GRAPH.SLOWLOG RESET graph_id
so it be subcommand?
can acl work like this?

File: src/slow_log/slow_log.c [hunk 5]
```diff
@@ -166,23 +212,66 @@ void SlowLog_Add(SlowLog *slowlog, const char *cmd, const char *query,
...
+	const SlowLog *slowlog,
+	RedisModuleCtx *ctx
+) {
 	SlowLog *aggregated_slowlog = SlowLog_New();
-	int my_t_id = ThreadPools_GetThreadID();
 
 	for(int t_id = 0; t_id < slowlog->count; t_id++) {
-		// Don't lock ourselves.
-		if(my_t_id != t_id) pthread_mutex_lock(slowlog->locks + t_id);
+		pthread_mutex_lock(slowlog->locks + t_id);
```
Comment (AviAvni) [inline src/slow_log/slow_log.c:272]: why we don't check if lock success?

File: tests/flow/test_slowlog.py [hunk 6]
```diff
@@ -59,3 +58,27 @@ def test_slowlog(self):
...
+        try:
+            slowlog = redis_con.execute_command("GRAPH.SLOWLOG", GRAPH_ID, "UNKNOW_SUB_CMD")
+        except ResponseError as e:
+            self.env.assertIn("Unknown subcommand", str(e))
+
+        # clear an existing slowlog
+        slowlog = redis_con.execute_command("GRAPH.SLOWLOG", GRAPH_ID)
+        self.env.assertGreater(len(slowlog), 0)
+
+        redis_con.execute_command("GRAPH.SLOWLOG", GRAPH_ID, "RESET")
```
Comment (AviAvni) [inline tests/flow/test_slowlog.py:79]: we don't add anything to the slow log to check that it cleared

File: tests/flow/test_slowlog.py [hunk 7]
```diff
@@ -59,3 +58,27 @@ def test_slowlog(self):
...
+        redis_con.execute_command("GRAPH.SLOWLOG", GRAPH_ID, "RESET")
```
Comment (AviAvni) [inline tests/flow/test_slowlog.py:80]: why duplicate reset?

**Review by AviAvni — CHANGES_REQUESTED** [review]

File: src/commands/cmd_slowlog.c (hunk 1 above)
Comment (swilly22) [reply src/commands/cmd_slowlog.c:29]: we always need to return `REDISMODULE_OK` even if we fail

File: src/commands/cmd_slowlog.c (hunk 2 above)
Comment (swilly22) [reply src/commands/cmd_slowlog.c:37]: we always need to return REDISMODULE_OK even if we fail

File: src/commands/cmd_slowlog.c (hunk 3 above)
Comment (swilly22) [reply src/commands/cmd_slowlog.c:50]: we always need to return REDISMODULE_OK even if we fail

File: docs/commands/graph.slowlog.md (hunk 4 above)
Comment (swilly22) [reply docs/commands/graph.slowlog.md:25]: I didn't want to create ambiguity with `GRAPH.SLOWLOG <key>`
the key is always at position `argv[1]`

File: tests/flow/test_slowlog.py (hunk 7 above)
Comment (swilly22) [reply tests/flow/test_slowlog.py:80]: making sure it's OK to reset an empty slowlog

File: docs/commands/graph.slowlog.md (hunk 4 above)
Comment (AviAvni) [reply docs/commands/graph.slowlog.md:25]: @MeirShpilraien help?

File: src/slow_log/slow_log.c [hunk 8]
```diff
@@ -166,23 +212,69 @@ void SlowLog_Add(SlowLog *slowlog, const char *cmd, const char *query,
...
+	const SlowLog *slowlog,
+	RedisModuleCtx *ctx
+) {
 	SlowLog *aggregated_slowlog = SlowLog_New();
-	int my_t_id = ThreadPools_GetThreadID();
 
 	for(int t_id = 0; t_id < slowlog->count; t_id++) {
-		// Don't lock ourselves.
-		if(my_t_id != t_id) pthread_mutex_lock(slowlog->locks + t_id);
+		if(pthread_mutex_lock(slowlog->locks + t_id) != 0) {
```
Comment (AviAvni) [inline src/slow_log/slow_log.c:272]: why lock in loop?

File: docs/commands/graph.slowlog.md (hunk 4 above)
Comment (AviAvni) [reply docs/commands/graph.slowlog.md:25]: I talked with @MeirShpilraien there are 2 options
1. introduce new command GRAPH.SLOWLOG2 that have 2 subcommands GRAPH.SLOWLOG2 GET ket GRAPH.SLOWLOG2 RESET key and deprecate GRAPH.SLOWLOG
2. introduce new command GRAPH.RESET_SLOWLOG key

Comment (AviAvni) [reply docs/commands/graph.slowlog.md:25]: from a talk with @gkorland and @LiorKogan we can leave it like this

**Pull request merged**